code/modules/vehicles/pioneer_stone.dm

```diff
@@ -0,0 +1,42 @@
+/obj/vehicle/ridden/pioneer_stone
+	name = "pioneer stone"
+	desc = "Pioneer's used to ride these babies for miles"
+	icon_state = "pioneer_stone"
+	max_integrity = 500
+	armor = list("melee" = 75, "bullet" = 0, "laser" = 25, "energy" = 25, "bomb" = -75, "bio" = 100, "rad" = 100, "fire" = 100, "acid" = 100) //it is a fucking stone, what do you expect
+	are_legs_exposed = TRUE
+	var/max_damage_force = 25
+	fall_off_if_missing_arms = TRUE
+	resistance_flags = FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF
+
+/obj/vehicle/ridden/pioneer_stone/Initialize()
+	. = ..()
+	var/datum/component/riding/D = LoadComponent(/datum/component/riding)
+	D.set_riding_offsets(RIDING_OFFSET_ALL, list(TEXT_NORTH = list(0, 4), TEXT_SOUTH = list(0, 4), TEXT_EAST = list(0, 4), TEXT_WEST = list( 0, 4)))
+	D.vehicle_move_delay = 2
+
+/obj/vehicle/ridden/pioneer_stone/Moved()
+	. = ..()
+	playsound(src, 'sound/effects/clang.ogg', 25, TRUE)
+
+/obj/vehicle/ridden/pioneer_stone/Bump(atom/movable/A)
+	. = ..()
+	if(A.density && has_buckled_mobs())
```

Comment:
this entire codeblock is nested to deep, use early returns so you dont end up 5 tabs indented